modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -160,6 +174,7 @@ const EditorUpload = function (editor: Editor): EditorUpload {
             editor.undoManager.transact(() => {
               Arr.each(imagesToRemove, (element) => {
                 editor.dom.remove(element);
+                replaceUrlInUndoStack(element.getAttribute('src'), Env.transparentSrc, replaceImageUrlForPlaceholder);
```

Comment:
This is the wrong place to do this, as this needs to be done before you attempt to create a new undo level. It should be done in the same place as a successful upload (eg line ~155)